eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -1199,30 +1207,83 @@ object Router {
     crc32c(data)
   }
 
-  case class ShortChannelIdsChunk(firstBlock: Long, numBlocks: Long, shortChannelIds: List[ShortChannelId])
+  case class ShortChannelIdsChunk(firstBlock: Long, numBlocks: Long, shortChannelIds: List[ShortChannelId]) {
+    /**
+     *
+     * @param maximumSize maximum size of the short channel ids list
+     * @return a chunk with at most `maximumSize` ids
+     */
+    def enforceMaximumSize(maximumSize: Int) = {
+      if (shortChannelIds.size <= maximumSize) this else {
+        // we use a random offset here, so even if shortChannelIds.size is much bigger than maximumSize (which should
+        // not happen) peers will eventually receive info about all channels in this chunk
+        val offset = Random.nextInt(shortChannelIds.size - maximumSize)
```

Comment:
off-by-one bug (reproduced with test provided below)

```suggestion
        val offset = Random.nextInt(shortChannelIds.size - maximumSize + 1)
```